### ISSUE TITLE
`vscode://settings/foo.bar` links do not work in chat window

### ISSUE THREAD

**Issue by bwateratmsft** [microsoft/vscode#229739]
Type: <b>Bug</b>

We have a need to refer to some settings in the chat responses. Ideally we could use the `[Link Notation](vscode://settings/setting.id)`, but this does not work in the chat response--for whatever reason the link is just rendered as plain text instead of a link.

Other links (like web links) work just fine so I'm not sure why it's not working.

VS Code version: Code - Insiders 1.94.0-insider (be2747e47e127ec674148b78dd9556c6317d18bd, 2024-09-25T05:04:31.141Z)
OS version: Windows_NT x64 10.0.22631
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i9-10900K CPU @ 3.70GHz (20 x 3696)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: enabled_on<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_graphite: disabled_off<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled<br>webnn: disabled_off|
|Load (avg)|undefined|
|Memory (System)|126.84GB (95.28GB free)|
|Process Argv|--log ms-azuretools.vscode-azure-github-copilot=debug --crash-reporter-id a639fe75-6727-46d4-bc6c-acdb3cb6366a|
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (19)</summary>

Extension|Author (truncated)|Version
---|---|---
tsl-problem-matcher|amo|0.6.2
esbuild-problem-matchers|con|0.0.3
vscode-eslint|dba|3.0.10
codespaces|Git|1.17.3
copilot|Git|1.234.1124
copilot-chat|Git|0.21.2024092501
azure-dev|ms-|0.8.3
vscode-azure-github-copilot|ms-|0.2.74
vscode-azureresourcegroups|ms-|0.9.5
vscode-docker|ms-|1.29.3
vscode-dotnet-runtime|ms-|2.1.6
debugpy|ms-|2024.11.2024092501
python|ms-|2024.15.2024092501
remote-containers|ms-|0.387.0
remote-wsl|ms-|0.88.4
extension-test-runner|ms-|0.0.12
vscode-selfhost-test-provider|ms-|0.3.25
vscode-yaml|red|1.15.0
fugio|Tyl|0.0.7

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vspor879:30202332
vspor708:30202333
vspor363:30204092
vswsl492cf:30256198
vscod805:30301674
vsaa593:30376534
py29gd2263:31024238
c4g48928:30535728
vscrpc:30624061
962ge761:30841072
pythongtdpath:30726887
welcomedialog:30812478
pythonnoceb:30776497
asynctok:30898717
dsvsc014:30777825
dsvsc015:30821418
pythonmypyd1:30859725
h48ei257:31000450
pythontbext0:30879054
accentitlementst:30870582
dsvsc016:30879898
dsvsc017:30880771
dsvsc018:30880772
cppperfnew:30980852
pythonait:30973460
945dj816:31013170
a69g1124:31018687
dvdeprecation:31040973
dwnewjupytercf:31046870
nb_pkg_only:31057982
nativerepl1:31134653
refactort:31084545
pythonrstrctxt:31093868
flighttreat:31119334
wkspc-onlycs-t:31132770
nativeloc1:31118317
wkspc-ranged-c:31125598
cf971741:31144450
e80f6927:31120813
fje88620:31121564
12bdf347:31141542
iacca1:31144502
notype1:31143044
c9j82188:31138334
showchatpanel:31139797
f8igb616:31140137

```

</details>

**Comment by isidorn** [comment]
Brandon wants our nice feature from release notes (rendering settings with a ⚙) also in the Chat view. Not sure if there is an underlying command they can trigger to open the setting, until we support rendering settings in chat (if we decide to do that)

**Comment by alexr00** [comment]
Yes, there's a command, example for the `git.openRepositoryInParentFolders` setting: `[git.openRepositoryInParentFolders](command:workbench.action.openSettings?%5B%22git.openRepositoryInParentFolders%22%5D)`. Maybe @roblourens can speak to why the `vscode://` link isn't working. 

The ⚙ work is only partially reusable because I didn't want to rewrite the whole release notes editor. Converting the markdown into the something that can be styled to look the like ⚙ thing and showing the context menu is reusable and is [here](https://github.com/microsoft/vscode/blob/9e23739d0b9f3381d99be9bde649709d4329daa5/src/vs/workbench/contrib/markdown/browser/markdownSettingRenderer.ts#L69-L88) and [here](https://github.com/microsoft/vscode/blob/9e23739d0b9f3381d99be9bde649709d4329daa5/src/vs/workbench/contrib/markdown/browser/markdownSettingRenderer.ts#L292-L307), but the [styling](https://github.com/microsoft/vscode/blob/9e23739d0b9f3381d99be9bde649709d4329daa5/src/vs/workbench/contrib/update/browser/releaseNotesEditor.ts#L276-L358) and the [click handling](https://github.com/microsoft/vscode/blob/9e23739d0b9f3381d99be9bde649709d4329daa5/src/vs/workbench/contrib/update/browser/releaseNotesEditor.ts#L395-L413) are not at all reusable.

Actually, the styles could probably be moved into each html element, then it's just the click handling that needs to be duplicated.
